startup/db.js

```diff
@@ -1,9 +1,15 @@
-const mongoose = require('mongoose');
-const winston = require('winston');
-const db_URI = 'mongodb://localhost:27017/MyUdemy';
-module.exports = function (){
-    //======== db setup ================
-    mongoose.connect(db_URI,{ useUnifiedTopology: true , useNewUrlParser: true})
-    .then(()=>winston.info('connected to mongodb'));
-    /// we don't need to catch the error the error will be catched from process.on in app.js
-}
\ No newline at end of file
+const mongoose = require('mongoose')
```

Comment:
* I'm setting the db URL here as a constant but i was planning to make it as environment variable later
* I'm using [express-async-errors](https://www.npmjs.com/package/express-async-errors) package to handle errors like this one here
* You used callback function to handle the connection result but the error is already handled by the error middleware and everything is logged using **Winston** package

Reply:
I understand what you are saying, but for some reason, this is not connecting to the db.  The winton logging is great, and very handy!! I'll see if i can resolve this problem... any suggestions to incorporate winston? (the change i made to this file connected to mongodb)

Reply:
honestly I don't get you, can you elaborate for me

Reply:
previously the way the file was set up there was no connection being made to the local mongoDB. The way i had rewritten the file connected to the local database so data was being added and could be retrieved from the mongoDB file.  
I have now changed the code back to how you had it, removing the export function and adding  "useCreateIndex: true".

The connection is now working to mongodb and a message is being added to the logfile.